### PR TITLE
Modify module_directive to be a supertype

### PR DESCRIPTION
**Pull request by berchn**
This is built with tree-sitter v0.20.4 and addresses ambiguities that
arise when analyzing a module_directive AST without the full context of
a CST.

This addresses https://github.com/tree-sitter/tree-sitter-java/issues/103.

Checklist:
- [ ] All tests pass in CI.
- [ ] There are sufficient tests for the new fix/feature.
- [ ] Grammar rules have not been renamed unless absolutely necessary.
- [ ] The conflicts section hasn't grown too much.
- [ ] The parser size hasn't grown too much (check the value of STATE_COUNT in src/parser.c).

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -644,23 +645,64 @@ module.exports = grammar({
 
     module_body: $ => seq(
       '{',
-      repeat($.module_directive),
+      repeat(
+        seq($.module_directive, ';')
```
Comment (maxbrunsfeld) [inline grammar.js:649]: Could you remove the `;` from here, and instead add it at the end of each type of module directive? It is helpful to have the `;` node grouped with the directive, similar to other declarations and statements.

**Review by maxbrunsfeld — COMMENTED** [review]
Thank you! This is definitely good change. I left one comment about a small change I'd like to see.

File: grammar.js (hunk 1 above)
Comment (berchn) [reply grammar.js]: I agree. Thanks for the catch.

**Comment by maxbrunsfeld** [comment]
Nice work @berchn ⚡

**Pull request merged**